src/main/java/core/basesyntax/ShopServiceImpl.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+    private final Map<String, Integer> storage = new HashMap<>();
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction : transactions) {
+            OperationHandler handler = operationStrategy.getHandler(transaction.getOperation());
+            handler.handle(transaction, storage);
+        }
```

Comment:
The process method correctly delegates the handling of transactions to the appropriate handler. However, consider adding a null check for the handler returned by operationStrategy.getHandler(). If there's no handler for a given operation, it should throw an informative RuntimeException to avoid NullPointerException.